crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -542,12 +494,46 @@ impl Persistence {
             !expired && !invalidated && !onchain_error && !fulfilled
         });
 
-        // Keep only relevant quotes.
-        next_quotes.retain(|uid, _quote| current_orders.contains_key(uid));
+        let new_quotes: HashMap<_, _> = {
+            let _timer = Metrics::get()
+                .database_queries
+                .with_label_values(&["read_quotes"])
+                .start_timer();
+
+            // Fetch quotes for only newly created and placed onchain orders.
+            let order_uids = current_orders
+                .values()
+                .filter_map(|order| {
+                    (order.metadata.onchain_user.is_some()
+                        || order.metadata.creation_date > after_timestamp)
+                        .then_some(ByteArray(order.metadata.uid.0))
+                })
+                .collect::<Vec<_>>();
+
+            database::orders::read_quotes(&mut tx, &order_uids)
+                .await?
+                .into_iter()
+                .filter_map(|quote| {
+                    let order_uid = domain::OrderUid(quote.order_uid.0);
+                    dto::quote::into_domain(quote)
+                        .map_err(|err| {
+                            tracing::warn!(?order_uid, ?err, "failed to convert quote from db")
+                        })
+                        .ok()
+                        .map(|quote| (order_uid, quote))
```

Comment:
nit: I think this makes it slightly easier to understand what's happening here due to named variables. No strong preference, though.
But I think we should use `inspect_err()` here since it's more idiomatic as we discard the error anyway.
```suggestion
                let quote = dto::quote::into_domain(quote)
                    .inspect_err(|err| {
                        tracing::warn!(?order_uid, ?err, "failed to convert quote from db")
                    })
                    .ok()?;
                Some((order_uid, quote))
```